### ISSUE TITLE
Parser error on valid test with variables

### ISSUE THREAD

**Issue by Julien-Elie** [koalaman/shellcheck#2455]
#### For bugs
- SC1072, SC1073
- 0.8.0
- [x] The rule's wiki page does not already cover this
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

VALUE=2
TST=eq
LIM=2

if [ "${VALUE}" "-${TST}" "${LIM}" ]; then
  echo OK
fi
```

#### Here's what shellcheck currently says:

```
In test line 7:
if [ "${VALUE}" "-${TST}" "${LIM}" ]; then
^-- SC1009 (info): The mentioned syntax error was in this if expression.
   ^-- SC1073 (error): Couldn't parse this test expression. Fix to allow more checks.
                  ^-- SC1072 (error): Expected a test operator. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

No error: neither a parsing error nor the expectation of a test operator.
